lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -0,0 +1,410 @@
+        func->GetFunctionProxy()->EnsureDeserialized()->SetDisplayName(methodName->GetString(), methodName->GetLength(), 0);
+
+        DynamicObject* prototype = GetPrototypeFromName(JavascriptOperators::GetPropertyId(className, scriptContext), scriptContext);
+        PropertyIds functionIdentifier = methodName->BufferEquals(_u("Symbol.iterator"), 15)? PropertyIds::_symbolIterator :
+            JavascriptOperators::GetPropertyId(methodName, scriptContext);
+
+        // Link the function to the prototype.
+        ScriptFunction* scriptFunction = library->CreateScriptFunction(func->GetFunctionProxy());
+        scriptFunction->SetIsJsBuiltInCode();
+        scriptFunction->GetFunctionProxy()->SetIsJsBuiltInCode();
+
+        if (forceInline)
+        {
+            Assert(scriptFunction->HasFunctionBody());
+            scriptFunction->GetFunctionBody()->SetJsBuiltInForceInline();
+        }
+        scriptFunction->SetPropertyWithAttributes(PropertyIds::length, TaggedInt::ToVarUnchecked(argumentsCount), PropertyConfigurable, nullptr);
+        scriptFunction->SetPropertyWithAttributes(PropertyIds::caller, library->nullValue, PropertyNone, nullptr);
+        scriptFunction->SetPropertyWithAttributes(PropertyIds::arguments, library->nullValue, PropertyNone, nullptr);
+
+        scriptFunction->SetConfigurable(PropertyIds::prototype, true);
+        scriptFunction->DeleteProperty(PropertyIds::prototype, Js::PropertyOperationFlags::PropertyOperation_None);
+
+        if (scriptFunction->GetScriptContext()->GetConfig()->IsES6FunctionNameEnabled())
+        {
+            scriptFunction->SetPropertyWithAttributes(PropertyIds::name, methodName, PropertyConfigurable, nullptr);
+        }
+
+        library->AddMember(prototype, functionIdentifier, scriptFunction);
+
+        if (!JavascriptOperators::IsUndefinedOrNull(aliasProperty))
+        {
+            JavascriptString * alias = JavascriptConversion::ToString(aliasProperty, scriptContext);
+            // Cannot do a string to property id search here, Symbol.* have different hashing mechanism, so resort to this str compare
+            PropertyIds aliasFunctionIdentifier = alias->BufferEquals(_u("Symbol.iterator"), 15) ? PropertyIds::_symbolIterator :
+                JavascriptOperators::GetPropertyId(alias, scriptContext);
+            library->AddMember(prototype, aliasFunctionIdentifier, scriptFunction);
+        }
+
+        if (functionIdentifier == PropertyIds::values)
+        {
+            library->arrayPrototypeDefaultValuesFunction = func;
+        }
+
+        //Don't need to return anything
+        return library->GetUndefined();
+    }
+
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_GetArrayLengthFunction(RecyclableObject *function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+
+        if (callInfo.Count < 2)
+        {
+            return scriptContext->GetLibrary()->GetUndefined();
```

Comment:
We don't expect this one right?, the count will always be greater then 2? Should we assert and return 0 instead of undefined?

Reply:
Right, converted to assert.